src/domain/auth-rebac/services/src/rebac_indexer.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use dill::{component, interface, meta};
+use init_on_startup::{InitOnStartup, InitOnStartupMeta};
+use internal_error::{InternalError, ResultIntoInternal};
+use kamu_accounts::{AccountService, JOB_KAMU_ACCOUNTS_PREDEFINED_ACCOUNTS_REGISTRATOR};
+use kamu_auth_rebac::{AccountPropertyName, DatasetPropertyName, RebacRepository, RebacService};
+use kamu_datasets::DatasetEntryService;
+use kamu_datasets_services::JOB_KAMU_DATASETS_DATASET_ENTRY_INDEXER;
+
+use crate::JOB_KAMU_REBAC_INDEXER;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct RebacIndexer {
+    rebac_repo: Arc<dyn RebacRepository>,
+    rebac_service: Arc<dyn RebacService>,
```

Comment:
Clearly, having both repo and service here means .. some operations are misplaced.

Reply:
Agreed. Only left `rebac_service`.

Done